src/libcore/any.rs

```diff
@@ -74,6 +74,16 @@ use crate::intrinsics;
 /// See the [module-level documentation][mod] for more details.
 ///
 /// [mod]: index.html
+// This trait is not unsafe, though we rely on the specifics of it's sole impl's
+// `type_id` function in unsafe code (e.g., `downcast`). Normally, that would be
+// a problem, but because the only impl of `Any` is a blanket implementation, no
+// other code can implement `Any`.
+//
+// We could plausibly make this trait unsafe -- it would not cause breakage,
+// since we control all the implementations -- but we choose not to as that's
+// both not really necessary and may confuse users about the distinction of
+// unsafe traits and unsafe methods (i.e., `type_id` would still be safe to call,
+// but we would likely want to indicate as such in documentation).
```

Comment:
To the contrary, I think making this trait unsafe will *help* explain the difference between unsafe traits and unsafe methods as it gives a fairly simple example for such a combination. Anyone writing unsafe code should be comfortable with that distinction.

Not a big deal, though.

Reply:
Yeah, fwiw, I had the same thought while reading this, but ultimately kept quiet cause "not a big deal". ^^

Reply:
I mostly just didn't want to wait for crater, but since people seem on board with unsafe here I might reconsider. I do think it's not a bad example.